homeassistant/components/tplink_omada/switch.py

```diff
@@ -208,11 +261,16 @@ async def async_added_to_hass(self) -> None:
     async def _async_turn_on_off(self, enable: bool) -> None:
         if self._port_details:
             self._port_details = await self.entity_description.set_func(
-                self.coordinator.omada_client, self.device, self._port_details, enable
+                self.coordinator.omada_client, self._device, self._port_details, enable
             )
-        self._attr_is_on = enable
-        # Refresh to make sure the requested changes stuck
-        await self.coordinator.async_request_refresh()
+
+        if self.entity_description.refresh_after_set:
```

Comment:
This kind of logic is an anti pattern when using entity descriptions. I'd pass both enable and port details and put a default for `update_func` that just returns enable.